packages/animations/lib/src/fade_transition.dart

```diff
@@ -0,0 +1,232 @@
+// Copyright 2019 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:animations/src/utils/curves.dart';
+import 'package:flutter/material.dart';
+
+/// Displays a modal above the current contents of the app.
+///
+/// This function displays the [FadeModalRoute], which transitions in
+/// with the Material fade transition.
+///
+/// Content below the modal is dimmed with a [ModalBarrier].
+///
+/// ```dart
+/// /// Sample widget that uses [showModalWithFadeTransition].
+/// class MyHomePage extends StatelessWidget {
+///   @override
+///   Widget build(BuildContext context) {
+///     return Scaffold(
+///       body: Center(
+///         child: RaisedButton(
+///           onPressed: () {
+///             showModalWithFadeTransition(
+///               context: context,
+///               child: FlutterLogoModal(),
+///             );
+///           },
+///           child: Icon(Icons.add),
+///         ),
+///       ),
+///     );
+///   }
+/// }
+///
+/// /// Displays a modal with the FlutterLogo on it.
+/// class FlutterLogoModal extends StatelessWidget {
+///   const _FlutterLogoModal();
+///
+///   @override
+///   Widget build(BuildContext context) {
+///     return Column(
```

Comment:
Why do we need the Column? Seems like it only has one child?

Reply:
Without the `Column` widget, the `ConstrainedBox` somehow does not respect the maximum width constraints and takes up the entire width of the screen instead. My guess is this is because it's the root widget of a new page, but I'm not entirely sure. I could investigate this and see if I could substitute in a widget that might make more sense here.

Reply:
That does seem odd. I would expect a `Center(child: SizedBox(...))` to work as expected. Can you show me what it looks like without?

Reply:
`SizedBox` worked, but `ConstrainedBox` didn't when I tried. I just swapped it out